lib/chef/knife/bootstrap_windows_base.rb

```diff
@@ -352,29 +335,63 @@ def bootstrap(proto=nil)
       def wait_for_remote_response(wait_max_minutes)
       end
 
+      def write_bootstrapper
+        write_command_file(render_template(load_template(config[:bootstrap_template])), bootstrap_bat_file)
+      end
+
       def bootstrap_command
         @bootstrap_command ||= "cmd.exe /C #{bootstrap_bat_file}"
       end
 
-      def bootstrap_render_banner_command(chunk_num)
-        "cmd.exe /C echo Rendering #{bootstrap_bat_file} chunk #{chunk_num}"
+      def bootstrap_render_banner_command(file_path, chunk_num)
+        "cmd.exe /C echo Rendering #{file_path} chunk #{chunk_num}"
+      end
+
+      def write_installer
+        opts = {}
+        unless config[:msi_url].nil? || config[:msi_url].empty?
+          opts[:install_msi_url] = config[:msi_url]
+        end
+        if config[:bootstrap_proxy]
+          opts[:http_proxy] = config[:bootstrap_proxy]
+          opts[:https_proxy] = config[:bootstrap_proxy]
+        end
+        opts[:prerelease] = config[:prerelease] if config[:prerelease]
+        puts "opts #{opts}"
+        write_command_file(
+          Mixlib::Install::ScriptGenerator.new(Chef::Config[:knife][:bootstrap_version] || true, true, opts).install_command,
```

Comment:
It's likely not well documented, but `Mixlib::Install::ScriptGenerator` will be deprecated in a future version. This is the original logic created for Kitchen. You'll want to instantiate `Mixlib::Install` with the appropriate options then call [#install_command](https://github.com/chef/mixlib-install/blob/master/lib/mixlib/install.rb#L81).